pkg/asset/installconfig/openstack/validation/machinepool.go

```diff
@@ -49,6 +49,10 @@ func ValidateMachinePool(p *openstack.MachinePool, ci *CloudInfo, controlPlane b
 			logrus.Warnf("Volume size is recommended to be greater than %d GB to use root volumes, had %d GB", recommendedStorage, p.RootVolume.Size)
 		}
 
+		if len(p.Zones) > 0 && len(p.RootVolume.Zones) == 0 {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("rootVolume").Child("zones"), p.RootVolume.Zones, "root volume availability zones must be specified when compute availability zones are specified"))
```

Comment:
The question I've asked myself is whether or not we want to enforce it only for the control plane machines or for all.

To me the same problem applies to workers which is why I've not made the difference - however I'm open for feedback on this one.

Reply:
Since this is a static validation, I think I have to move it to `pkg/types/openstack/validation/machinepool.go`. I learnt that today thanks to @pierreprinetti .

I'll address that in a next patchset.

Reply:
Nice, thanks